lib/diaspora_federation/entity.rb

```diff
@@ -264,21 +315,52 @@ def simple_node(doc, name, value)
       end
     end
 
-    # @param [Nokogiri::XML::Element] root_node xml nodes
+    def json_data
+      enriched_properties.map {|key, value|
+        type = self.class.class_props[key]
+
+        if type.instance_of?(Class) && value.respond_to?(:to_json_hash)
+          entity_data = value.to_json_hash[:entity_data]
+          [key, entity_data] unless entity_data.nil?
+        else
+          [key, value]
+        end
```

Comment:
We don't "depend" on ActiveSupport, but we depend on something calling `to_json` again on elements in an array (while having hashes for single nested entities). So I don't like the fact, that we still have "raw" entities in the result of `to_json` and not a clean hash.